### PR TITLE
feat: filter features by type

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Can filter search results by one or many feature types.
Based on this example it's easy to add other filter types. The important part of research here was handling multiple values in query params.

### CODE REVIEW

File: src/lib/openapi/spec/feature-search-query-parameters.ts [hunk 1]
```diff
@@ -18,11 +18,23 @@ export const featureSearchQueryParameters = [
...
+            },
+        },
+        description: 'The list of feature types to filter by',
         in: 'query',
     },
 ] as const;
 
-export type FeatureSearchQueryParameters = FromQueryParams<
-    typeof featureSearchQueryParameters
+export type FeatureSearchQueryParameters = Partial<
```
Comment (kwasniew) [inline src/lib/openapi/spec/feature-search-query-parameters.ts:38]: All query params are optional

File: src/lib/features/feature-toggle/types/feature-toggle-strategies-store-type.ts [hunk 2]
```diff
@@ -19,6 +19,14 @@ export interface FeatureConfigurationClient {
     variants: IVariant[];
     dependencies?: IDependency[];
 }
+
+export interface IFeatureSearchParams {
```
Comment (kwasniew) [inline src/lib/features/feature-toggle/types/feature-toggle-strategies-store-type.ts:23]: This type mixes data from query params and userId so decided to create a custom type that is used by store and service

File: src/lib/features/feature-search/feature.search.e2e.test.ts [hunk 3]
```diff
@@ -59,7 +66,21 @@ test('should return matching features by name', async () => {
     });
 });
 
-test('should return matching features by tag', async () => {
+test('should filter features by type', async () => {
+    await app.createFeature({ name: 'my_feature_a', type: 'release' });
+    await app.createFeature({ name: 'my_feature_b', type: 'experimental' });
+
+    const { body } = await filterFeaturesByType([
```
Comment (kwasniew) [inline src/lib/features/feature-search/feature.search.e2e.test.ts:73]: showing one matching type and one that is not used

File: src/lib/features/feature-search/feature-search-controller.ts [hunk 4]
```diff
@@ -71,10 +71,13 @@ export default class FeatureSearchController extends Controller {
         res: Response,
     ): Promise<void> {
         if (this.config.flagResolver.isEnabled('featureSearchAPI')) {
-            const { query = '', projectId = '' } = req.query;
+            const { query, projectId, type } = req.query;
```
Comment (kwasniew) [inline src/lib/features/feature-search/feature-search-controller.ts:74]: no need for default values

**Review by andreas-unleash — APPROVED** [review]
Nice work. LGTM!

**Pull request merged**